File: apps/api/src/Booking.Domain/Entities/Tenancy/Plan.cs
using Booking.Domain.Enums;

namespace Booking.Domain.Entities.Tenancy;

public class Plan : Idendity.Entity
{
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public PlanType PlanType { get; set; }
    public decimal Price { get; set; }
    public string BillingCycle { get; set; } = "Monthly"; // Monthly, Yearly
    
    // Plan Features
    public int MaxWorkers { get; set; }
    public int MaxServices { get; set; }
    public int MaxReservationsPerMonth { get; set; }
    public bool HasCustomBranding { get; set; }
    public bool HasAnalytics { get; set; }
    public bool HasApiAccess { get; set; }
    
    // Navigation Properties
    public ICollection<TenantPlan> TenantPlans { get; set; } = new List<TenantPlan>();
}

File: apps/web/src/services/plan.ts
import { apiClient } from './api';

export interface PlanDto {
  id: string;
  name: string;
  description?: string;
  planType: number; // 0: Free, 1: Basic, 2: Professional, 3: Enterprise
  price: number;
  billingCycle?: string; // e.g. 'monthly' | 'yearly'
  maxWorkers?: number;
  maxServices?: number;
  maxReservationsPerMonth?: number;
  hasCustomBranding?: boolean;
  hasAnalytics?: boolean;
  hasApiAccess?: boolean;
  created?: string;
  lastModified?: string;
}

export const planService = {
  /**
   * Returns available plans from the backend.
   */
  async getPlans(): Promise<PlanDto[]> {
    return apiClient.get<PlanDto[]>('/plans');
  }
};
